crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -106,10 +116,24 @@ impl CumulocityConverter {
 
     fn try_convert_event(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> {
         let tedge_event = ThinEdgeEvent::try_from(input.topic.name.as_str(), input.payload_str()?)?;
-        let smartrest_alarm = event::serialize_event(tedge_event)?;
-        let smartrest_topic = Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC);
+        dbg!(input.payload_bytes().len());
+        if input.payload_bytes().len() > MQTT_EVENT_SIZE_LIMIT {
+            let event_text = tedge_event
+                .data
+                .and_then(|data| data.message)
+                .unwrap_or_else(|| "generic event".into());
+            let _ = block_on(self.http_proxy.send_event(
+                tedge_event.name.as_str(),
+                event_text.as_str(),
+                None,
+            ))?;
```

Comment:
We are already inside an `async` context and it is pretty much impossible to spawn a another executor (doable but dangerous and unnecessarily complicated), even though getting the runtime with current will succeed it is may fail anyway. 
The safest and pretty much recommended way to do it is just to convert the calling function to be an `async`, which I think you already had tried.

Theoretically this could be possible but it is dangerous as it blocks the entire runtime, so I would rather go with marking the function async and execute the future on main runtime.         
```rust 
let handle = Handle::current();
handle.enter();
futures::executor::block_on(...)
```

Reply:
This is more of a temporary placeholder code and will be be replaced with a simple await call like:

```
self.http_proxy.send_event(
                tedge_event.name.as_str(),
                event_text.as_str(),
                None,
            ).await?;
```

once #789, where `Converter::try_convert` is made async, is merged.